File: oxygen-cms-api/src/main/java/com/ladbrokescoral/oxygen/cms/api/entity/OnBoardingGuide.java
package com.ladbrokescoral.oxygen.cms.api.entity;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import javax.validation.constraints.NotBlank;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.mongodb.core.mapping.Document;

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
@Document(collection = "onBoardingGuides")
@Data
@EqualsAndHashCode(callSuper = true)
public class OnBoardingGuide extends SortableEntity implements HasBrand {

  @NotBlank private String brand;
  @NotBlank private String guideName;
  private String guidePath;
  private SvgFilename svgFilename;
  private boolean enabled;
}

File: oxygen-cms-ui/src/app/client/private/models/onBoardingGuide.ts
import {Base} from '@app/client/private/models/base.model';
import {SvgFilename} from '@app/client/private/models/svgfilename.model';

export interface OnBoardingGuide extends Base {
  svgFilename: SvgFilename;
  guideName: string;
  guidePath: string;
  enabled: boolean;
}
